conformance_tests/sysman/test_sysman_vf_management/src/test_sysman_vf_management.cpp

```diff
@@ -0,0 +1,439 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#include "test_harness/test_harness.hpp"
+
+namespace lzt = level_zero_tests;
+
+namespace {
+
+#ifdef USE_ZESINIT
+class VfManagementZesTest : public lzt::ZesSysmanCtsClass {
+public:
+  bool is_vf_enabled = false;
+};
+#define VF_MANAGEMENT_TEST VfManagementZesTest
+#else // USE_ZESINIT
+class VfManagementTest : public lzt::SysmanCtsClass {
+public:
+  bool is_vf_enabled = false;
+};
+#define VF_MANAGEMENT_TEST VfManagementTest
+#endif // USE_ZESINIT
+
+void validate_vf_capabilities(zes_vf_exp_capabilities_t &vf_capabilities,
+                              uint32_t &vf_count) {
+  EXPECT_GE(vf_capabilities.address.domain, 0);
+  EXPECT_LE(vf_capabilities.address.domain, MAX_DOMAINs);
+  EXPECT_GE(vf_capabilities.address.bus, 0);
+  EXPECT_LE(vf_capabilities.address.bus, MAX_BUSES_PER_DOMAIN);
+  EXPECT_GE(vf_capabilities.address.device, 0);
+  EXPECT_LE(vf_capabilities.address.device, MAX_DEVICES_PER_BUS);
+  EXPECT_GE(vf_capabilities.address.function, 0);
+  EXPECT_LE(vf_capabilities.address.function, MAX_FUNCTIONS_PER_DEVICE);
+  EXPECT_GT(vf_capabilities.vfDeviceMemSize, 0);
+  EXPECT_GT(vf_capabilities.vfID, 0);
+  EXPECT_LE(vf_capabilities.vfID, vf_count);
+}
+
+void compare_vf_capabilities(zes_vf_exp_capabilities_t &vf_capability_initial,
+                             zes_vf_exp_capabilities_t &vf_capability_later) {
+  EXPECT_EQ(vf_capability_initial.stype, vf_capability_later.stype);
+  EXPECT_EQ(vf_capability_initial.pNext, vf_capability_later.pNext);
+  EXPECT_EQ(vf_capability_initial.address.domain,
+            vf_capability_later.address.domain);
+  EXPECT_EQ(vf_capability_initial.address.bus, vf_capability_later.address.bus);
+  EXPECT_EQ(vf_capability_initial.address.device,
+            vf_capability_later.address.device);
+  EXPECT_EQ(vf_capability_initial.address.function,
+            vf_capability_later.address.function);
+  EXPECT_EQ(vf_capability_initial.vfDeviceMemSize,
+            vf_capability_later.vfDeviceMemSize);
+  EXPECT_EQ(vf_capability_initial.vfID, vf_capability_later.vfID);
+}
+
+TEST_F(
+    VF_MANAGEMENT_TEST,
+    GivenValidDeviceWhenCallingEnumEnabledVFExpThenValidVFHandlesAreReturned) {
+  for (auto device : devices) {
+    uint32_t count = lzt::get_vf_handles_count(device);
+    if (count > 0) {
+      is_vf_enabled = true;
+      LOG_INFO << "VF is enabled on this device!!";
+      auto vf_handles = lzt::get_vf_handles(device, count);
+
+      for (const auto &vf_handle : vf_handles) {
+        EXPECT_NE(vf_handle, nullptr);
+      }
+    } else {
+      LOG_INFO << "No VF handles found for this device!";
+    }
+  }
+
+  if (!is_vf_enabled) {
```

Comment:
Is this condition required ?

Reply:
yes. This is added to track whether vf is enabled on at least one device. After iterating through all devices, if vf is not enabled on any of the devices, then failing the test